ts/src/time-profiler-inspector.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TimeProfile, TimeProfileNode} from './v8-types';
+import * as inspector from 'node:inspector';
+
+const session = new inspector.Session();
+session.connect();
+
+// Wrappers around inspector functions
+export function startProfiling(): Promise<void> {
+  return new Promise<void>((resolve, reject) => {
+    session.post('Profiler.enable', err => {
+      if (err !== null) {
+        reject(err);
+      }
+      session.post('Profiler.start', err => {
```

Comment:
There's no option to pass runName (title) in this API. Is this problematic for overlapping profiles?

- https://nodejs.org/api/inspector.html#cpu-profiler_1
- calls this dev tools protocol method https://chromedevtools.github.io/devtools-protocol/v8/Profiler/#method-enable

Reply:
Overlapping profiles will never happen if a user only has the Node.js profiling agent running (since the agent will collect a profile before asking the backend if it can collect another). 

This difference is likely fine.

Reply:
For now I'm just removing the name param altogether. If we get FRs in the future, we can add this session mechanism.